File: engine/engine-model/src/main/scala/server/life/OverrideMonsterStats.scala
package server.life

class OverrideMonsterStats(var hp: Int, var mp: Int, var exp: Int, private var _change: Boolean) {
  def this(hp: Int, mp: Int, exp: Int) = {
    this(hp, mp, exp, true)
  }

  def this() = {
    this(1, 0, 0, true)
  }
}

File: engine/engine-model/src/main/java/server/life/OverrideMonsterStats.java
package server.life;

public class OverrideMonsterStats {
   private final Integer hp;

   private final Integer mp;

   private final Integer exp;

   private final Boolean change;

   public OverrideMonsterStats(Integer hp, Integer mp, Integer exp, Boolean change) {
      this.hp = hp;
      this.mp = mp;
      this.exp = exp;
      this.change = change;
   }

   public OverrideMonsterStats(Integer hp, Integer mp, Integer exp) {
      this(hp, mp, exp, true);
   }

   public OverrideMonsterStats() {
      this(1, 0, 0, true);
   }

   public Integer hp() {
      return hp;
   }

   public Integer mp() {
      return mp;
   }

   public Integer exp() {
      return exp;
   }

   public Boolean change() {
      return change;
   }
}
